test/lib/RedefineClassHelper.java

```diff
@@ -61,6 +64,64 @@ public static void redefineClass(Class<?> clazz, byte[] bytecode) throws Excepti
         instrumentation.redefineClasses(new ClassDefinition(clazz, bytecode));
     }
 
+    private static byte[] getBytecodes(ClassLoader loader, String name) throws Exception {
+        InputStream is = loader.getResourceAsStream(name + ".class");
```

Comment:
I think this should be 
try(InputStream is = loader.getResourceAsStream(name + ".class");) {
...

Reply:
yes, only change is that stream is closed.

Reply:
That's really cool.  Thanks for the suggestion.